docs/design/2022-06-30-partition-table-batchwrite-suppport.md

```diff
@@ -0,0 +1,145 @@
+
+The semantics to be supported in this feature are:
+
+- INSERT semantics with Data Source API
+- REPLACE semantics with Data Source API
+- DELETE semantics with Spark SQL
+
+And the associated supported partition types to be supported in this feature are:
+
+- [Range partitioning](https://docs.pingcap.com/tidb/dev/partitioned-table#range-partitioning)
+- [Hash partitioning](https://docs.pingcap.com/tidb/dev/partitioned-table#hash-partitioning)
+
+And the parititon expression must be under contions as bellow:
++ column expression
++ `YEAR` (expression) where the expression is a column and its type is datetime or string literal
+that can be parsed as datetime.
+
+## Motivation and background
+
+Partitioning is a way in which a database splits its actual data down into separate tables, but
+still gets treated as a single table by the SQL layer.
+The main goal is to reduce the amount of data read for particular SQL operations so that overall
+response time is reduced while it also can be used to accomplish a number of various objectives.
+
+### The feature-level implementation of partitioning in TiDB
+
+Currently, TiDB supports Range partitioning, List partitioning, List COLUMNS partitioning, and Hash
+partitioning.
+For a table partitioned by [RANGE COLUMNS](https://dev.mysql.com/doc/refman/5.7/en/partitioning-columns-range.html),
+currently TiDB only supports using a single partitioning column.
+You can read the [documentation](https://docs.pingcap.com/tidb/dev/partitioned-table#compatibility-with-mysql) for more
+details.
+
+### The code-level implementation of partitioning in TiDB
+
+If a table is partitioned,
+the [`TableInfo`](https://github.com/pingcap/tidb/blob/v6.1.0/parser/model/model.go#L370-L447) holds
+the id of the single table by the SQL layer, which can be called logical tableId,
+while the n(n is the number of partitions)
+[`PartitionDefinition`](https://github.com/pingcap/tidb/blob/v6.1.0/parser/model/model.go#L1076-L1084)
+each holds the id of the table stored in TiKV, which can be called physical tableId.
+
+It's typical that the logical tableId should be used to allocate id or gain meta information, while
+the physical tableId should be used to generate encoded key or any scans which we operate on the
+physical table.
+
+## Goals
+
+Since TiSpark supports INSERT, REPLACE semantics with DataSource API and DELETE statement with Spark SQL, 
+it's natural to support the partitioning feature in these semantics.
+
+List partitioning, List COLUMNS partitioning are generally available in V6.1 recently and also more
+complicated. Considering the workload of this feature, we have decide to support them in the future.
+In this feature, we only focus on Range partitioning and Hash partitioning.
+
+Currently TiSpark only support YEAR function, so the parititon expression must be under contions as bellow:
++ column expression
++ `YEAR` (expression) where the expression is a column and its type is datetime or string literal
```

Comment:
Shouldn't it be Year(fcuntion) here?

Reply:
Year() is the function,  expression is the arguments of the function.
I will change it to the argument to avoid misunderstanding.